src/storage/checkpoint_manager.cpp

```diff
@@ -375,38 +375,39 @@ void CheckpointReader::ReadSequence(ClientContext &context, Deserializer &deseri
 //===--------------------------------------------------------------------===//
 // Indexes
 //===--------------------------------------------------------------------===//
-void CheckpointWriter::WriteIndex(IndexCatalogEntry &index_catalog, Serializer &serializer) {
-	// The index data is written as part of WriteTableData.
-	// Here, we need only serialize the pointer to that data.
-	auto root_block_pointer = index_catalog.index->GetRootBlockPointer();
-	serializer.WriteProperty(100, "index", &index_catalog);
-	serializer.WriteProperty(101, "root_block_pointer", root_block_pointer);
+void CheckpointWriter::WriteIndex(IndexCatalogEntry &index_catalog_entry, Serializer &serializer) {
+	// The index data is written as part of WriteTableData
+	// Here, we serialize the index catalog entry
+
+	// we need to keep the tag "index", even though it is slightly misleading
+	serializer.WriteProperty(100, "index", &index_catalog_entry);
+
+	// also, we write a dummy BlockPointer, because when reading the file, we don't know if we're
+	// reading an older storage version (with BlockPointers) or not
+	serializer.WriteProperty(101, "root_block_pointer", BlockPointer());
```

Comment:
If you want to delete this property we can remove the write, and on the read have a default value (`BlockPointer()`) using `ReadPropertyWithDefault`